deep_agent/src/infrastructure/backend.py

```
@@ -69,6 +69,113 @@ def upload_files(self, files: list[tuple[str, bytes]]) -> list[FileUploadRespons
+
+        deduped_facts = [f for i, f in enumerate(facts) if i not in indices_to_remove]
+        result_lines = non_fact_lines + [f"- {f}" for f in deduped_facts]
+        logger.debug(
+            "Deduplicated memory: %d facts → %d (removed %d)",
+            len(facts),
+            len(deduped_facts),
+            len(indices_to_remove),
+        )
+        return "\n".join(result_lines) + "\n"
+
+    def _is_memory_path(self, file_path: str) -> bool:
```

Ruff (0.16.3)

[warning] Unused method argument: `file_path`

(ARG002)